monitor.cpp

```diff
@@ -4,34 +4,53 @@
 #include <chrono>
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
-
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
+void pattern(std::string pat) {
+cout << pat << flush;
+sleep_for(seconds(1));
+}
+void report(std::string entry)
+{
+  cout<<entry;
+}
+void displayMessage(std::string msg) {
+    report(msg);
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        pattern("\r*");
+        pattern("\r *");
     }
+}
+
+
+bool criticalTemp(float temperature)
+{
+  if(temperature > 102 || temperature < 95)
+    return true;
+  else
+    return false;
+}
+bool criticalPulseRate(float pulseRate)
+{
+ if (pulseRate < 60 || pulseRate > 100)
```

Comment:
Though the tool does not point it out, the range check is similar here, like in the `criticalTemp`. So for example, if the range semantics change (like `<` becomes `<=`), the change will need to be done in multiple places.
Try isolating the range check to another function.